lib/engine/game/g_1849/stock_market.rb

```diff
@@ -38,7 +38,14 @@ def up(corporation, coordinates)
 
         def right(corporation, coordinates)
           price = share_price(coordinates).price
-          return super if !BLOCKED_RIGHT_PRICES.include?(price) || @game.phase.status.include?('blue_zone')
+
+          # Lower two blocked right prices push stock marker UP if not in blue zone
+          if !@game.phase.status.include?('blue_zone') && BLOCKED_RIGHT_PRICES.slice(0, 2).include?(price)
+            @game.log << "#{corporation.name} share price blocked from moving right by phase; moves up instead" if corporation
```

Comment:
These methods are used for more than stock price movement, so the log message shouldn't go here. If you feel a message is needed (I think it's okay without one), then the message would go in the `move` method.

The other log message below is a bug.